pilot/pkg/proxy/envoy/v2/debug_test.go

```diff
@@ -30,69 +31,98 @@ func Test_Syncz(t *testing.T) {
+	})
+	t.Run("sync status not set when Nackd", func(t *testing.T) {
+		initLocalPilotTestEnv(t)
+		adsstr := connectADS(t, util.MockPilotGrpcAddr)
+		defer adsstr.CloseSend()
+
+		sendEDSReq(t, []string{"outbound|9080||app2.default.svc.cluster.local"}, sidecarId(app3Ip, "syncApp2"), adsstr)
+		sendEDSNack(t, []string{"outbound|9080||app2.default.svc.cluster.local"}, sidecarId(app3Ip, "syncApp2"), adsstr)
+		sendCDSReq(t, sidecarId(app3Ip, "syncApp2"), adsstr)
+		sendCDSNack(t, sidecarId(app3Ip, "syncApp2"), adsstr)
+		sendLDSReq(t, sidecarId(app3Ip, "syncApp2"), adsstr)
+		sendLDSNack(t, sidecarId(app3Ip, "syncApp2"), adsstr)
+		sendRDSReq(t, sidecarId(app3Ip, "syncApp2"), []string{"80", "8080"}, adsstr)
+		sendRDSNack(t, sidecarId(app3Ip, "syncApp2"), []string{"80", "8080"}, adsstr)
+		for i := 0; i < 5; i++ {
+			_, err := adsReceive(adsstr, 5*time.Second)
+			if err != nil {
+				t.Fatal("Recv failed", err)
 			}
-			if ss.ClusterSent != "" {
-				checkMap["cSent"] = true
+		}
+		node, _ := model.ParseServiceNode(sidecarId(app3Ip, "syncApp2"))
+		verifySyncStatus(t, getSyncStatus(t), node.ID, true, false)
+	})
+}
+
+func getSyncStatus(t *testing.T) []v2.SyncStatus {
+	req, err := http.NewRequest("GET", "/debug", nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+	rr := httptest.NewRecorder()
+	syncz := http.HandlerFunc(v2.Syncz)
+	syncz.ServeHTTP(rr, req)
+	got := []v2.SyncStatus{}
+	if err := json.Unmarshal(rr.Body.Bytes(), &got); err != nil {
+		t.Error(err)
+	}
+	return got
+}
+
+func verifySyncStatus(t *testing.T, gotStatus []v2.SyncStatus, nodeID string, wantSent, wantAcked bool) {
+	// This is a mostly horrible hack because the single pilot instance is shared across multiple tests
+	// This makes this test contaminated by others
+	for _, ss := range gotStatus {
+		if ss.ProxyID == nodeID {
+			if (ss.ClusterSent != "") != wantSent {
+				t.Errorf("wanted ClusterSent set %v got %v for %v", wantSent, ss.ClusterSent, nodeID)
 			}
-			if ss.ClusterAcked != "" {
-				checkMap["cAck"] = true
+			if (ss.ClusterAcked != "") != wantAcked {
+				t.Errorf("wanted ClusterAcked set %v got %v for %v", wantAcked, ss.ClusterAcked, nodeID)
 			}
-			if ss.ListenerSent != "" {
-				checkMap["lSent"] = true
+			if (ss.ListenerSent != "") != wantSent {
+				t.Errorf("wanted ListenerSent set %v got %v for %v", wantSent, ss.ListenerSent, nodeID)
 			}
-			if ss.ListenerAcked != "" {
-				checkMap["lAck"] = true
+			if (ss.ListenerAcked != "") != wantAcked {
+				t.Errorf("wanted ListenerAcked set %v got %v for %v", wantAcked, ss.ListenerAcked, nodeID)
 			}
-			if ss.RouteSent != "" {
-				checkMap["rSent"] = true
+			if (ss.RouteSent != "") != wantSent {
+				t.Errorf("wanted RouteSent set %v got %v for %v", wantSent, ss.RouteSent, nodeID)
 			}
-			if ss.RouteAcked != "" {
-				checkMap["rAck"] = true
+			if (ss.RouteAcked != "") != wantAcked {
+				t.Errorf("wanted RouteAcked set %v got %v for %v", wantAcked, ss.RouteAcked, nodeID)
```

Comment:
Looks like this change made the daily build flakey.

Example: https://k8s-gubernator.appspot.com/build/istio-prow/pull/istio-releases_daily-release/793/daily-unit-tests/384/

```
debug_test.go:157: wanted RouteAcked set true got  for syncApp-644fc65469-96dza.testns
```